apps/api/src/app/inbox/usecases/mark-many-notifications-as/mark-many-notifications-as.usecase.ts

```diff
@@ -63,4 +75,57 @@ export class MarkManyNotificationsAs {
       groupId: subscriber._organizationId,
     });
   }
+
+  private async logTraces(command: MarkManyNotificationsAsCommand, _subscriberId: string): Promise<void> {
+    const messages = await this.messageRepository.find({
+      _environmentId: command.environmentId,
+      _subscriberId,
+      _id: { $in: command.ids },
+    });
+
+    for (const message of messages) {
+      try {
+        if (!message._jobId) continue;
+
+        if (command.read !== undefined) {
+          await this.traceLogRepository.create(
+            createTraceLog(message, command, command.read ? 'message_read' : 'message_unread')
+          );
+        }
+
+        if (command.snoozedUntil !== undefined) {
+          await this.traceLogRepository.create(createTraceLog(message, command, 'message_snoozed'));
+        }
+
+        if (command.archived !== undefined) {
+          await this.traceLogRepository.create(
+            createTraceLog(message, command, command.archived ? 'message_archived' : 'message_unarchived')
+          );
+        }
+      } catch (error) {
+        this.logger.warn({ err: error }, `Failed to create engagement trace for message ${message._id}`);
+      }
+    }
+  }
+}
+
+function createTraceLog(
+  message: MessageEntity,
+  command: MarkManyNotificationsAsCommand,
+  eventType: TraceEvent
+): Omit<Trace, 'id' | 'expires_at'> {
+  return {
+    created_at: LogRepository.formatDateTime64(new Date()),
+    organization_id: message._organizationId,
+    environment_id: message._environmentId,
+    user_id: command.subscriberId,
+    subscriber_id: message._subscriberId,
```

Comment:
I'm not sure i like this :( maybe internal subscriber id and external? or internal and just subscriber id? Just user id is not something we have in our language at all today